### PR TITLE
adding/fixing documentation and figures

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
@gwaybio This PR is ready for review!

### CODE REVIEW

**Comment by jenna-tomkinson** [comment]
@axiomcura @phicks22 

This PR is adding more documentation to the NF1 project, specifically around the IC step. 

I would like for feedback on the content and any issues within it (e.g. grammar, spelling, etc.). 

No coding changed FYI

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Images/
```
Comment (axiomcura) [inline .gitignore:1]: I don't think `.gitignore` files should be present in repos.

File: 1_preprocessing_data/README.md [hunk 2]
```diff
@@ -2,13 +2,37 @@
...
 
-## Illumination Correction
+## Illumination Correction (IC)
 
 To correct for illumination issues within the NF1 data, we use the BaSiC method that was established in an article by [Peng et al.](https://doi.org/10.1038/ncomms14836).
 We specifically use the Python implementation of this method, called [PyBaSiC](https://github.com/peng-lab/BaSiCPy).
 
 Illumination correction is an important step in cell image analysis pipelines as it helps with downstream processes like segmentation (more accuracy in segmenting shape and identifying objects to segment) and feature extraction (accurate measurements in intensity, texture, etc.).
 
+Being able to visualize if images need IC can be very easy or very hard depending on the dataset.
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]:
```suggestion
To visualize if images need IC can be simple or difficult depending on the dataset. 
```

File: 1_preprocessing_data/README.md [hunk 3]
```diff
@@ -2,13 +2,37 @@
...
-## Illumination Correction
+## Illumination Correction (IC)
 
 To correct for illumination issues within the NF1 data, we use the BaSiC method that was established in an article by [Peng et al.](https://doi.org/10.1038/ncomms14836).
 We specifically use the Python implementation of this method, called [PyBaSiC](https://github.com/peng-lab/BaSiCPy).
 
 Illumination correction is an important step in cell image analysis pipelines as it helps with downstream processes like segmentation (more accuracy in segmenting shape and identifying objects to segment) and feature extraction (accurate measurements in intensity, texture, etc.).
 
+Being able to visualize if images need IC can be very easy or very hard depending on the dataset. 
+In the NF1 dataset, the raw images are impossible to tell with the human eye that there are any illumination issues.
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]:
```suggestion
In the NF1 dataset, Identifying illumination issues within raw images are impossible to capture with the human eye that .
```

File: 1_preprocessing_data/README.md [hunk 4]
```diff
@@ -2,13 +2,37 @@
...
+
+To be able to see illumination issues within the images, we use Fiji to increase the maximum limits of the display range to brighten the image using the `B&C` function (Figure 1).
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]:
```suggestion
To capture these illumination issues, we used Fiji to increase the maximum limits of the display range to brighten the images using the `B&C` function (Figure 1).
```
I do not know if `image` should be plural here.

File: 1_preprocessing_data/README.md [hunk 5]
```diff
@@ -2,13 +2,37 @@
...
+
+![NF1_IC_Demonstration.png](example_figs/NF1_IC_Demonstration.png)
+> Figure 1. Use of Fiji to Determine Illumination Issues. This figure shows how increasing the maximum limit for the display range reveals illumination issues (e.g. vignetting like in the DAPI channel). The increasing of the upper limits for display starts to show where there is brighter areas in the image that was hidden when the upper limit was lower.
+
+The documentation regarding this function states:
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]: Maybe I missed it while reading it.

Would it be best to add the name of the function? Fiji is a well known program! Adding the name will make it easier for other readers/users to find which function you are talking about!

File: 1_preprocessing_data/README.md [hunk 6]
```diff
@@ -2,13 +2,37 @@
...
+
+> ImageJ displays images by linearly mapping pixel values in the display range to display values in the range 0-255. Pixels with a value less than the minimum are displayed as black and those with a value greater than the maximum are displayed as white. `Minimum` and `Maximum` control the lower and upper limits of the display range.
+
+After running PyBaSiC and outputting corrected images, Fiji can be used once again to be able to visualize if the illumination issues been corrected (Figure 2).
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]:
```suggestion
After running PyBaSiC and generating corrected images, Fiji is executed again to ensure if the illumination issues have been corrected (Figure 2).
```

File: 1_preprocessing_data/README.md [hunk 7]
```diff
@@ -2,13 +2,37 @@
...
+
+![NF1_IC_Brightened_Fig.png](example_figs/NF1_IC_Brightened_Fig.png)
+> Figure 2. Visualizing IC Through Comparison. This figure shows how impactful illumination correction using PyBaSiC is on the raw images. 
+
+Without using Fiji, the raw and corrected images look almost exactly the same and there is no way to tell (with the human eye) if the illumination issues have been corrected. (Figure 3).
```
Comment (axiomcura) [inline 1_preprocessing_data/README.md]:
```suggestion
Without using Fiji, the raw and corrected images almost look identical and it is difficult to tell (with the human eye) whether the illumination issues have been corrected. (Figure 3).
```

**Review by axiomcura — COMMENTED** [review]
Hey @jenna-tomkinson  I have added some suggestions to your documentation. Feel free to accept or reject the changes!

File: .gitignore (hunk 1 above)
Comment (jenna-tomkinson) [reply .gitignore:1]: @gwaybio @axiomcura 

Greg, in our last meeting I had asked about if the .gitignore should be ignored or added and you said that .gitignores should be present within the repos. Can you please confirm this?

Comment (gwaybio) [reply .gitignore:1]: Yes, we include gitignore in analysis repos https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files

File: 1_preprocessing_data/README.md (hunk 4 above)
Comment (jenna-tomkinson) [reply 1_preprocessing_data/README.md]: :+1:

File: 1_preprocessing_data/README.md (hunk 5 above)
Comment (jenna-tomkinson) [reply 1_preprocessing_data/README.md]: @axiomcura I just realized I am using the wrong term for it, technically it is a "tool". 

I also named it above this figure which says:

> To visualize the illumination issues, we use Fiji to increase the maximum limits of the display range to brighten the images using the `B&C` function (Figure 1).

I will update the documentation to call it a tool and not a function.

**Comment by jenna-tomkinson** [comment]
@gwaybio I have made changes to this PR using Erik's suggestions. Can you please review so we can merge? :smiley:

File: 1_preprocessing_data/README.md [hunk 8]
```diff
@@ -2,13 +2,37 @@
...
 
 Illumination correction is an important step in cell image analysis pipelines as it helps with downstream processes like segmentation (more accuracy in segmenting shape and identifying objects to segment) and feature extraction (accurate measurements in intensity, texture, etc.).
 
+Visualizing if images need IC can be simple or difficult depending on the dataset. 
+In the NF1 dataset, identifying illumination issues is impossible to capture with the human eye.
+
+To visualize the illumination issues, we use Fiji to increase the maximum limits of the display range to brighten the images using the `B&C` tool (Figure 1).
+
+![NF1_IC_Demonstration.png](example_figs/NF1_IC_Demonstration.png)
+> Figure 1. Use of Fiji to Determine Illumination Issues. This figure shows how increasing the maximum limit for the display range reveals illumination issues (e.g. vignetting like in the DAPI channel). The increasing of the upper limits for display starts to show where there is brighter areas in the image that was hidden when the upper limit was lower.
```
Comment (gwaybio) [inline 1_preprocessing_data/README.md]:
```suggestion
> Figure 1. Use of Fiji to determine illumination issues. This figure shows how increasing the maximum limit for the display range reveals illumination issues (e.g. vignetting like in the DAPI channel). Increasing upper limits shows brighter areas in the image that were hidden previously.
```

edited for lower case and understanding. Please check me!

File: 1_preprocessing_data/README.md [hunk 9]
```diff
@@ -2,13 +2,37 @@
...
+
+The documentation regarding this tool states:
+
+> ImageJ displays images by linearly mapping pixel values in the display range to display values in the range 0-255. Pixels with a value less than the minimum are displayed as black and those with a value greater than the maximum are displayed as white. `Minimum` and `Maximum` control the lower and upper limits of the display range.
+
+After running PyBaSiC and generating corrected images, Fiji is used to confirm if the illumination issues been corrected (Figure 2).
```
Comment (gwaybio) [inline 1_preprocessing_data/README.md]:
```suggestion
After running PyBaSiC and generating corrected images, we use Fiji to confirm that the illumination issues been corrected (Figure 2).
```

Active voice :)

File: 1_preprocessing_data/README.md [hunk 10]
```diff
@@ -2,13 +2,37 @@
...
+
+![NF1_IC_Brightened_Fig.png](example_figs/NF1_IC_Brightened_Fig.png)
+> Figure 2. Visualizing IC Through Comparison. This figure shows how impactful illumination correction using PyBaSiC is on the raw images.
```
Comment (gwaybio) [inline 1_preprocessing_data/README.md]:
```suggestion
> Figure 2. Visualizing IC through comparison. This figure shows the impact of IC using PyBaSiC on the raw images. 
```

Capitalization, abbreviation, and brevity.

File: 1_preprocessing_data/README.md [hunk 11]
```diff
@@ -2,13 +2,37 @@
...
+
+After running PyBaSiC and generating corrected images, Fiji is used to confirm if the illumination issues been corrected (Figure 2).
+
+![NF1_IC_Brightened_Fig.png](example_figs/NF1_IC_Brightened_Fig.png)
+> Figure 2. Visualizing IC Through Comparison. This figure shows how impactful illumination correction using PyBaSiC is on the raw images. 
+
+Without using Fiji, the raw and corrected images look almost identical and it is very difficult to tell (with the human eye) if the illumination issues have been corrected. (Figure 3).
+
+![NF1_IC_No_Brightening.png](example_figs/NF1_IC_No_Brightening.png)
+> Figure 3. Raw versus Corrected Images. This figure shows how only looking at the images without any editing does not give any indication if the IC worked or not.
```
Comment (gwaybio) [inline 1_preprocessing_data/README.md]:
```suggestion
> Figure 3. Raw versus corrected images. This figure shows how only looking at the images without any editing does not give any indication if the IC worked or not.
```

Capitalization

File: 2_segmenting_data/README.md [hunk 12]
```diff
@@ -1,4 +1,4 @@
-# Segment Schwann Cells from NF1 Data
+# 2. Segment NF1 Data
```
Comment (gwaybio) [inline 2_segmenting_data/README.md]: Why remove Schwann cells? I thought this was important - maybe you talk about it later in this readme tho.

**Review by gwaybio — APPROVED** [review]
LGTM! The figures look beautiful.

A couple text-based comments for clarity/consistency.

File: 2_segmenting_data/README.md (hunk 12 above)
Comment (jenna-tomkinson) [reply 2_segmenting_data/README.md]: The goal was to make it follow the other modules by adding the `2.`,  but I totally agree the "Schwann Cells" part is important and I don't mention it later.

File: 1_preprocessing_data/README.md (hunk 8 above)
Comment (jenna-tomkinson) [reply 1_preprocessing_data/README.md]: That is a much more concise way of putting it! Fixed!

File: 1_preprocessing_data/README.md (hunk 9 above)
Comment (jenna-tomkinson) [reply 1_preprocessing_data/README.md]: One day I will get over the passive voice habit from college! Ty!

**Comment by jenna-tomkinson** [comment]
@gwaybio I will merge this now! Ty!

**Pull request merged**